### PR TITLE
Synchronous Kafka producer

### PR DESCRIPTION
**Pull request by AValsamis**
## Which problem is this PR solving?

This PR is a proposed solution for #36.

## Short description of the changes

- Implemented a synchronous kafka producer component with tracing support.

Since this implementation depends on sarama.RecordHeader, that supports kafka versions greater than v11.0, the component is available for those versions only. 

- Introduced an example that uses the new component.

### CODE REVIEW

File: examples/sixth/main.go [hunk 1]
```diff
@@ -0,0 +1,125 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"github.com/beatlabs/patron/trace/sync/kafka"
```
Comment (mantzas) [inline examples/sixth/main.go:6]: import order is off

File: trace/sync/kafka/kafka.go [hunk 2]
```diff
@@ -0,0 +1,124 @@
+package kafka
```
Comment (mantzas) [inline trace/sync/kafka/kafka.go:1]: I think it makes more sense to have the old import path:

trace/kafka/sync
trace/kafka/async

and 

trace/kafka could contains common code

File: trace/sync/kafka/kafka.go [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+
+import (
+	"context"
+	"github.com/beatlabs/patron/trace"
```
Comment (mantzas) [inline trace/sync/kafka/kafka.go:5]: ordering again

**Review by mantzas — CHANGES_REQUESTED** [review]

File: trace/sync/kafka/kafka.go [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+	"github.com/beatlabs/patron/encoding/json"
+	"github.com/beatlabs/patron/errors"
+	"github.com/opentracing/opentracing-go"
+	"github.com/opentracing/opentracing-go/ext"
+)
+
+// Message abstraction of a Kafka message.
+type Message struct {
+	topic string
+	body  []byte
```
Comment (drakos74) [inline trace/sync/kafka/kafka.go:17]: should we maybe also add the key ?
see https://github.com/beatlabs/patron/pull/86 ?

File: examples/second/main.go [hunk 5]
```diff
@@ -3,6 +3,7 @@ package main
 import (
 	"context"
 	"fmt"
+	"github.com/beatlabs/patron/trace/async/kafka"
```
Comment (sotirispl) [inline examples/second/main.go:6]: I think you should run goimports to re-order the imports

**Comment by sotirispl** [comment]
You have some conflicts. Could you please resolve them along with the other CR comments?

**Comment by gm42** [comment]
@AValsamis do you mind if I pick this up? I can create another PR picking your commits and then solve the conflicts and process the review feedback.

**Comment by AValsamis** [comment]
> @AValsamis do you mind if I pick this up? I can create another PR picking your commits and then solve the conflicts and process the review feedback.

Sure, you are more than welcome to! Apologies, I meant to pick this up these last few weeks but just couldn't find the time.

File: trace/sync/kafka/kafka.go (hunk 4 above)
Comment (gm42) [reply trace/sync/kafka/kafka.go:17]: @drakos74 I have covered it in https://github.com/beatlabs/patron/pull/170

**Comment by gm42** [comment]
My PR to cover this functionality is here: https://github.com/beatlabs/patron/pull/170

**Pull request closed without merging**